modules/client/src/main/java/org/apache/ignite/internal/client/ReliableChannel.java

```diff
@@ -606,6 +621,21 @@ private void initAllChannelsAsync() {
         );
     }
 
+    private void onTopologyAssignmentChanged(ClientChannel clientChannel) {
+        // NOTE: Multiple channels will send the same update to us, resulting in multiple cache invalidations.
+        // This could be solved with a cluster-wide AssignmentVersion, but we don't have that.
```

Comment:
Let's select a single server that will be used to monitor changes in assignment. We have heartbeats anyway so we will get updates from it even if we do not use it for communication.

Reply:
That's a great idea! Done. `testClientReceivesPartitionAssignmentUpdates` updated to check heartbeat scenario.

Reply:
Updated the IEP too https://cwiki.apache.org/confluence/display/IGNITE/IEP-95+Client+Partition+Awareness